### PR TITLE
chore: adding new gtm tag for unlocking publishing (request)

### PR DESCRIPTION
**Pull request by daine**
# Summary | Résumé

Adds new data-* class selector for google tag manager (gtm) to trigger on Unlock Publishing requests

# Pull Request Checklist

Please complete the following items in the checklist before you request a review:

- [x] Have you completely tested the functionality of change introduced in this PR? Is the PR solving the problem it's meant to solve within the scope of the related issue?
- [x] The PR does not introduce any new issues such as failed tests, console warnings or new bugs.
- [ ] If this PR adds a package have you ensured its licensed correctly and does not add additional security issues?
- [x] Is the code clean, readable and maintainable? Is it easy to understand and comprehend.
- [ ] Does your code have adequate comprehensible comments? Do new functions have [docstrings](https://tsdoc.org/)?
- [ ] Have you modified the change log and updated any relevant documentation?
- [ ] Is there adequate test coverage? Both unit tests and end-to-end tests where applicable?
- [ ] If your PR is touching any UI is it accessible? Have you tested it with a screen reader? Have you tested it with automated testing tools such as axe?

### CODE REVIEW

File: components/forms/Button/Button.tsx [hunk 1]
```diff
@@ -48,6 +52,7 @@ export const Button = (
       data-testid={testid ? testid : "button"}
       disabled={disabled}
       {...ariaProps}
+      {...(dataAttrName && dataAttrValue ? { [`data-${dataAttrName}`]: dataAttrValue } : "")}
```
Comment (timarney) [inline components/forms/Button/Button.tsx:55]: I would prefer we use `...props` at this point vs adding more props to the args.

We'll be looking to re-factor the Button component in the future to remove some of these "extra" props i.e. shape that have been added.

We'll also look to remove standard props that a html button should just handle like data attributes, disabled etc...

Comment (daine) [reply components/forms/Button/Button.tsx]: I think there are 2 Button components (there might be a purpose to having two? not sure), the other one also had this [multiple extra prop issue](https://github.com/cds-snc/platform-forms-client/blob/develop/components/globals/Buttons/Button.tsx).

I modified the PR to remove extras, let me know if that works or if it needs changing. Thanks @timarney!

**Comment by timarney** [comment]
Will review and get into the release upcoming on Mon.

File: components/forms/Button/Button.tsx [hunk 2]
```diff
@@ -12,23 +11,14 @@ interface ButtonProps {
   destructive?: boolean;
   testid?: string;
   ariaProps?: React.AriaAttributes;
+  dataAttrName?: string;
```
Comment (timarney) [inline components/forms/Button/Button.tsx:14]: Given the `...props` we should be able to remove the `dataAttrName` and  `dataAttrValue`

File: components/forms/Button/Button.tsx [hunk 3]
```diff
@@ -12,23 +11,14 @@ interface ButtonProps {
   destructive?: boolean;
   testid?: string;
   ariaProps?: React.AriaAttributes;
```
Comment (timarney) [inline components/forms/Button/Button.tsx:13]: Looks like `ariaProps` was removed but we still have it here.

File: components/forms/Button/Button.tsx (hunk 1 above)
Comment (timarney) [reply components/forms/Button/Button.tsx]: Right --- yes there are 2 buttons components 

The "original" and the one being used on the page your modifying is the button for "public forms" vs the other one that we don't need to modify for this PR is a button we created for use within the form builder / admin.  At the time we didn't want to alter anything with "public forms".  At some point once we get some styling sorted out we can re-combine but that's way outside the scope of this PR :)

File: components/forms/Button/Button.tsx (hunk 2 above)
Comment (daine) [reply components/forms/Button/Button.tsx]: Removed!

File: components/forms/Button/Button.tsx [hunk 4]
```diff
@@ -11,24 +10,12 @@ interface ButtonProps {
...
-    secondary,
-    base,
-    onClick,
-    className,
-    testid,
-    destructive,
-    disabled,
-    ariaProps,
-  } = props;
+  const { children, secondary, base, onClick, className, testid, destructive } = props;
```
Comment (timarney) [inline components/forms/Button/Button.tsx:18]: Looking at the failing test ... 
 
 we should use ...rest 
 
 { children, secondary, base, onClick, className, testid, destructive, ...rest } = props;
 
 ```
 <button
      className={classes}
      onClick={onClick}
      data-testid={testid ? testid : "button"}
      {...rest}
    >
 ```

**Review by timarney — APPROVED** [review]

**Pull request merged**